contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -138,7 +138,7 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
      * @dev See {AccessControl-_revokeRole}.
      */
     function _revokeRole(bytes32 role, address account) internal virtual override {
-        if (role == DEFAULT_ADMIN_ROLE && account == _currentDefaultAdmin) {
```

Comment:
My original rationale for using `_currentDefaultAdmin` instead of `defaultAdmin()` is that `_revokeRole()` is internal and thus, it's less obvious that an override will affect it. I think also @frangio flagged something similar in the original PR (#4009).

I'd revert this one unless we have a very good reason for changing it. The other one is good under my original rationale because `renounceRole` is public.

Reply:
>it's less obvious that an override will affect it

Do you mean that for the developer that overrides `defaultAdmin` it is less obvious that the override will affect `_revokeRole`? I'm not convinced by this argument, I think the consistency is more valuable.

Reply:
>it may be excessive using the public methods everywhere

Yes, I recall saying this, it was not in this line. I still feel that way... But the solution that I see is to remove virtual, as long as we have virtual we have to more or less use the overrideable functions.